src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadataBuilder.cs

```diff
@@ -39,10 +45,11 @@ public class ClonedPackageSearchMetadata : IPackageSearchMetadata
             public string Title { get; set; }
             public bool PrefixReserved { get; set; }
             public LicenseMetadata LicenseMetadata { get; set; }
-            public PackageDeprecationMetadata DeprecationMetadata { get; set; }
 
-            public AsyncLazy<IEnumerable<VersionInfo>> LazyVersionsFactory { get; set; }
+            internal AsyncLazy<PackageDeprecationMetadata> LazyDeprecationFactory { get; set; }
+            public async Task<PackageDeprecationMetadata> GetDeprecationMetadataAsync() => await LazyDeprecationFactory;
 
+            internal AsyncLazy<IEnumerable<VersionInfo>> LazyVersionsFactory { get; set; }
             public async Task<IEnumerable<VersionInfo>> GetVersionsAsync() => await LazyVersionsFactory;
```

Comment:
How do you guarantee `LazyVersionsFactory` is not `null`?

Reply:
Using `??` now with a static readonly lazy.